jemalloc-ctl/src/error.rs

```diff
@@ -0,0 +1,92 @@
+//! Error type
+
+use libc::c_int;
+use {fmt, result};
+
+/// Error of the `jemalloc_sys::mallct`-family of functions.
+///
+/// The `jemalloc-sys` crate: `mallctl`, `mallctlnametomib`, and `mallctlbymib``
+/// functions return `0` on success; otherwise they return an error value.
+#[derive(Copy, Clone, PartialEq)]
+#[repr(i32)]
+pub enum Error {
```

Comment:
Are you sure this is the universe of all possible error codes that jemalloc will ever return? It might be safer to make this a wrapper over an i32 and use associated constants rather than an enum:

```rust
pub struct Error(i32);

impl Error {
    pub const EINVAL: Error = Error(libc::EINVAL);
    // ...
}
```

Reply:
> Are you sure this is the universe of all possible error codes that jemalloc will ever return? 

Obviously no, I don't know if `malloctl` will ever return some other error codes. What I know is that these are the errors being returned right now, and that all future error codes will be `NonZero`, so I could add a `Unknown(NonZero<c_int>)` variant to remain forward compatible while still keeping the `size_of::<Result<(), Error>() == size_of::<c_int>()` optimization.

Reply:
@sfackler I've replaced this with a `struct Error(NonZero<c_int>)`.